store/store.go

```diff
@@ -74,20 +77,27 @@ func NewStore(params *Parameters, basePath string) (*Store, error) {
 		return nil, fmt.Errorf("creating empty file: %w", err)
 	}
 
-	recentEDSCache, err := cache.NewAccessorCache("recent", params.RecentBlocksCacheSize)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create recent eds cache: %w", err)
+	var recentEDSCache cache.Cache = cache.NoopCache{}
+	if params.RecentBlocksCacheSize > 0 {
+		recentEDSCache, err = cache.NewAccessorCache("recent", params.RecentBlocksCacheSize)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create recent eds cache: %w", err)
+		}
 	}
 
-	availabilityCache, err := cache.NewAccessorCache("availability", params.AvailabilityCacheSize)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create availability cache: %w", err)
+	var availabilityCache cache.Cache = cache.NoopCache{}
+	if params.AvailabilityCacheSize > 0 {
+		availabilityCache, err = cache.NewAccessorCache("availability", params.AvailabilityCacheSize)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create availability cache: %w", err)
+		}
 	}
 
 	store := &Store{
-		basepath:  basePath,
-		cache:     cache.NewDoubleCache(recentEDSCache, availabilityCache),
```

Comment:
It looks like you removed DoubleCache, but still read from both caches every time. I wonder what is the reason to remove, because double cache did exactly the same, while also having advantage of having read logic in single place instead of multiple copies.

Reply:
There is only one time where Get happens. There other time happens because there is no Has method on the Cache. Anyway, I will add private Get method to avoid duplication until we get Has.

Reply:
Why not return double cache that does exactly what is needed?